workflows/checks/check-md5.nf

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+// run parameters
+params.run_metafile = 's3://ccdl-scpca-data/sample_info/scpca-library-metadata.tsv'
+params.outdir = 's3://nextflow-ccdl-results/scpca-checks/'
+
+process check_md5{
+  container 'ubuntu:20.04'
+  publishDir "${params.outdir}"
+  input:
+    tuple val(id), val(md5_file), path(files)
```

Comment:
Sort of... this is a super strange syntax, I think, partly because the inclusion/exclusion of parenthesis seems strange.

This is more like the definition of parameters for a function. In `R`, this might look like
```
function(id, md5_file, files){}
```
But it is a bit stranger than that, because really the first argument expects a tuple, so it is more like:
```
function(my_tuple){
id = my_tuple[1]
md5_file = my_tuple[2]
files = my_tuple[3]
...
}
```

In other words, this is actually defining where we will put the values for `id`, `md5_file` and `files`. The actual assignment comes when we call the process later with `check_md5(ch_runs)`.  So `ch_runs` will pass a tuple with three values, which will then get assigned to `id`, `md5_file` and `files`.

`id` and `md5_file` will be assigned as a `val` variable type (strings in this case), where `files` is a `path`, which lets nextflow know that there are files associated with it. We need to tell nextflow that so it knows that it will need to get those files to run the process. 

(An odd thing here is that `md5_file` is also a file in fact, so you would think it needs to be gotten too, but it is one of the files in `files` and we aren't allowed to tell nextflow to get the same file twice.)

Reply:
Ahh interesting, it does make much more sense now! So essentially once nextflow gets the files in `files`, each file in `files` becomes a `val` variable type.

Reply:
Sort of? I think they stay a list of `paths`, but functionally we can mostly treat them the same. There is some weirdness where the actual location of the file in question is not something we are supposed to care about. I ran into this when I made assumptions that turned out to be incorrect, but I don't remember the details.